bigquery/samples/tests/conftest.py

```diff
@@ -40,9 +40,9 @@ def random_dataset_id(client):
     random_dataset_id = "example_dataset_{}_{}".format(
         now.strftime("%Y%m%d%H%M%S"), uuid.uuid4().hex[:8]
     )
-    random_dataset = client.create_dataset(random_dataset_id)
-    yield "{}.{}".format(client.project, random_dataset.random_table_id)
-    client.delete_dataset(random_dataset, delete_contents=True, not_found_ok=True)
+    dataset = client.create_dataset(random_dataset_id)
+    client.delete_dataset(dataset, delete_contents=True, not_found_ok=True)
+    yield "{}.{}".format(client.project, dataset.dataset_id)
```

Comment:
Delete needs to come after yield.

Learn more about what this is doing here: https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code

Reply:
I see you reverted this change, thanks. I do recommend reading that pytest documentation section to understand fixtures better.

Reply:
Thank you for the link!